src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -32,19 +33,16 @@ public class CryptoConfig
 
         private const string ECDsaIdentifier = "ECDsa";
 
-        private static volatile Dictionary<string, string> s_defaultOidHT = null;
-        private static volatile Dictionary<string, object> s_defaultNameHT = null;
-        private static volatile Dictionary<string, Type> appNameHT = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
-        private static volatile Dictionary<string, string> appOidHT = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private static volatile Dictionary<string, string> s_defaultOidHT;
+        private static volatile ConcurrentDictionary<string, object> s_defaultNameHT;
```

Comment:
now we will have 3 locks meaning that each of the dictionaries is thread safe but overall state might be torn, @bartonjs is that something we should be concerned about?

Reply:
s_defaultNameHT is only ever built once (though with this change it looks like now it upgrades itself for delay-loaded type references).

It's actually a bit unfortunate that it needs to be a ConcurrentDictionary at all, but if a major point of this change is to overwrite the strings with Type objects for forward-references then this is the way to go.

But since that upgrade is just a memoization cache on non-user controlled data, there's no state to really be torn, it's "eventually consistent".

Reply:
sounds good - I'm not familiar with the code, AddAlgorithm here suggested there might be more here than just the cache